openmetadata-service/src/main/java/org/openmetadata/service/config/S3Configuration.java

```diff
@@ -0,0 +1,49 @@
+package org.openmetadata.service.config;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.NotBlank;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.lang3.StringUtils;
+
+@Getter
+@Setter
+public class S3Configuration {
```

Comment:
Could we reuse this somehow in here? https://github.com/open-metadata/OpenMetadata/blob/main/openmetadata-spec/src/main/resources/json/schema/security/credentials/awsCredentials.json#L7

We might just need to add the bucket and keep centralized creds management.

Same would be true for Azure/GCP.

Reply:
we might be able to configure the supported storages in JSON Schema and reuse those defs in Collate for the factories etc. as well